### PR TITLE
fix(frontend): Error when API key is not entered is not clear

### LINKED ISSUE #4236: [Bug]: Error when API key is not entered is not clear
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

In the frontend, if no API key has been entered, we should have a comprehensible error. However, if not API key is entered, we get an opaque error that just says "An error occurred while running the agent"

image (2)
image (1)

This should be made more clear. @amanape , could you take a look at this?

### Current OpenHands version

```bash
0.9
```

### Installation and Configuration

```bash
docker?
```

### Model and Agent

_No response_

### Operating System

_No response_

### Reproduction Steps

_No response_

### Logs, Errors, Screenshots, and Additional Context

See conversation on slack: https://openhands-ai.slack.com/archives/C06U8UTKSAD/p1728265897191799

### PR DESCRIPTION
**Pull request by Yashwanth-Chandrakumar**
Give a summary of what the PR does, explaining any non-trivial design decisions

Added a modal for prompting the user to input an API Key in settings during startup and in session.

Link of any specific issues this addresses

Resolves #4236

### CODE REVIEW

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape is this good

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape i have solved the issue but why isnt it merged

**Comment by amanape** [comment]
I will give a proper review soon, sorry for the delay! In the meantime, I see that there are some linting issues on the frontend. Could you take care of it to pass the GitHub CI?

**Comment by Yashwanth-Chandrakumar** [comment]
Yes i think the prettier in my code editor is causing the problem will solve it. @amanape

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape solved the lint issues perfect to merge

File: frontend/src/components/form/settings-form.tsx [hunk 1]
```diff
@@ -106,7 +114,31 @@ export function SettingsForm({
       submitForm(formData);
     }
   };
+  const handleCloseClick = () => {
+    console.log("Close clicked, hasEnteredKey:", hasEnteredKey);
```
Comment (amanape) [inline frontend/src/components/form/settings-form.tsx:118]: Remove the `console.log`

File: frontend/src/components/form/settings-form.tsx [hunk 2]
```diff
@@ -106,7 +114,31 @@ export function SettingsForm({
...
+    if (!hasEnteredKey) {
+      console.log("Showing warning modal");
```
Comment (amanape) [inline frontend/src/components/form/settings-form.tsx:120]: Remove the `console.log`

File: frontend/src/components/form/settings-form.tsx [hunk 3]
```diff
@@ -106,7 +114,31 @@ export function SettingsForm({
...
+      setShowWarningModal(true);
+    } else {
+      console.log("Closing settings");
```
Comment (amanape) [inline frontend/src/components/form/settings-form.tsx:123]: Remove the `console.log`

File: frontend/src/components/form/settings-form.tsx [hunk 4]
```diff
@@ -106,7 +114,31 @@ export function SettingsForm({
...
+    onClose();
+  };
+
+  const handleWarningCancel = () => {
+    setShowWarningModal(false);
+  };
+
+  const handleApiKeyChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = !!event.target.value;
+    console.log("API key changed, new value:", newValue);
```
Comment (amanape) [inline frontend/src/components/form/settings-form.tsx:139]: Remove the `console.log`

**Review by amanape — COMMENTED** [review]
The functionality does not work when pressing "Save". The flow should look something like this:

- User presses "Save"
- If there is no API key, show modal
- Otherwise, send the save settings action request (it currently only does this)

We are using Remix to handle form actions. Instead of keeping `LLM_API_KEY` changes in the `hasEnteredKey` state, we can access this setting from the `formRef`:

```ts
const formData = new FormData(formRef.current); // assuming formRef.current is not undefined
const apiKey = formData.get("api-key");
```

You should prefer using Remix / React Router (latest version) API to handle form data and navigation states. I recommend reading their documentation for more information!

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape ok i will implement the changes and come back

**Comment by Yashwanth-Chandrakumar** [comment]
Hi @amanape corrected the flow and other eslint issues (removed all the console log statements).

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape all checks are done can we merge it

**Review by amanape — CHANGES_REQUESTED** [review]
There seems to be just one bug:

1. Set API key and save
2. Open settings again
3. Clear API key input and press "Save"
4. Confirm warning modal

In this case, the API key is not actually removed, and still there.

**Comment by Yashwanth-Chandrakumar** [comment]
@amanape solved the bug, ready to merge

**Review by amanape — APPROVED** [review]
Thanks for the changes! Bonus points for handling the zIndex, too.

**Pull request merged**